browser/ui/views/brave_actions/brave_shields_action_view.cc

```diff
@@ -56,32 +60,13 @@ BraveShieldsActionView::BraveShieldsActionView()
   SetHasInkDropActionOnClick(true);
   SetHorizontalAlignment(gfx::ALIGN_CENTER);
   ink_drop->SetVisibleOpacity(kToolbarInkDropVisibleOpacity);
+  tab_strip_model_->AddObserver(this);
```

Comment:
Needs corresponding `RemoveObserver(this)`.  - `ScopedObservation`

Reply:
NVM - `TabStripModelObserver` already included scoped observation functionality.
Just `AddObser` seems fine here.

Reply:
TabStripModelObserver doesnt need to be explicitly removed. It does it, implicitly.